src/test/java/org/sonatype/nexus/ci/iq/ClassFilterLoggingTestTrait.groovy

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.iq
+
+import java.util.logging.Level
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowExecution
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.jvnet.hudson.test.LoggerRule
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * Add this trait to any test that should have an extra verification that no JEP-200 marshalling errors have occurred.
+ * See INT-2407 and https://docs.sonatype.com/x/op5NCQ for additional information and INT-418 for the original work.
+ */
+trait ClassFilterLoggingTestTrait
+{
+  Logger logger = LoggerFactory.getLogger(ClassFilterLoggingTestTrait)
+
+  @Rule
+  LoggerRule loggerRule = new LoggerRule()
+
+  def helpfulMessage = '''A possible JEP-200 marshalling error has occurred. It is likely that additional classes 
+will need to be added to the 'hudson.remoting.ClassFilter' file. See https://docs.sonatype.com/x/op5NCQ'''
```

Comment:
I assume this'll put the message out in 2 lines when it's used?  Maybe it would be cleaner to transform it right back into a single line: ```...'''.replaceAll('\n', ' ')`?

Reply:
Or use `.stripMargin`, which is probably the more common way to do this.

Reply:
Yes 2 lines. This only shows up in junit output so only ever seen by us and hopefully we'll never encounter JEP-200 again thus never even see it. Point being its likely not worth any back and forth. Plus, if you really want one line then you'll have to scroll horizontally and you'll wonder why the original dev didn't split it into two lines! :rofl:  :roll_eyes: